modules/features/discover/src/main/java/au/com/shiftyjelly/pocketcasts/discover/view/CarouselItemViewHolder.kt

```diff
@@ -56,23 +62,25 @@ class CarouselItemViewHolder(val theme: Theme, itemView: View) : RecyclerView.Vi
         lblTagline.chipBackgroundColor = ColorStateList.valueOf(ThemeColor.contrast04(theme.activeTheme))
     }
 
-    var podcast: DiscoverPodcast? = null
-        set(value) {
-            field = value
-            if (value != null) {
-                imageRequestFactory.createForPodcast(value.uuid).loadInto(imageView)
-                lblTitle.text = value.title
-                lblTitle.setTextColor(ThemeColor.contrast01(theme.activeTheme))
-                lblSubtitle.text = value.author
-                lblSubtitle.setTextColor(ThemeColor.contrast03(theme.activeTheme))
-                setBackingGradient(value.color)
-                btnSubscribe.updateSubscribeButtonIcon(
-                    subscribed = value.isSubscribed,
-                    colorSubscribed = UR.attr.contrast_02,
-                    colorUnsubscribed = UR.attr.contrast_02,
-                )
-            } else {
-                reset()
+    fun setPodcast(podcast: DiscoverPodcast?, isTablet: Boolean, isRankedList: Boolean = false) {
+        if (podcast != null) {
+            imageRequestFactory.createForPodcast(podcast.uuid).loadInto(imageView)
+            lblSubtitle.setTextColor(ThemeColor.contrast03(theme.activeTheme))
+            setBackingGradient(podcast.color)
+            btnSubscribe.updateSubscribeButtonIcon(
+                subscribed = podcast.isSubscribed,
+                colorSubscribed = UR.attr.contrast_02,
+                colorUnsubscribed = UR.attr.contrast_02,
+            )
+            lblTitle.text = podcast.title
+            lblTitle.setTextColor(ThemeColor.contrast01(theme.activeTheme))
+            lblSubtitle.text = podcast.author
+
+            if (lblTitle.text.length > MAX_TITLE_SIZE && !isTablet && !isRankedList) {
+                lblTitle.setTextSize(TypedValue.COMPLEX_UNIT_SP, 20F)
             }
```

Comment:
Wouldn't `TextViewCompat.setAutoSizeTextTypeUniformWithConfiguration()` more suitable for this?